.github/workflows/ci.yml

```diff
@@ -162,3 +162,27 @@ jobs:
 
       - name: Download dependencies
         run: go mod download && go mod tidy
+
+  e2e_test:
+    runs-on: ubuntu-latest
+    needs:
+      - test
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Install Go
```

Comment:
We can setup the Go caches to reduce the time to prepare a environment. For example,
```
      - name: Get Go environment
        id: go-env
        run: |
          echo "::set-output name=cache::$(go env GOCACHE)"
          echo "::set-output name=modcache::$(go env GOMODCACHE)"
      - name: Set up cache
        uses: actions/cache@v3
        with:
          path: |
            ${{ steps.go-env.outputs.cache }}
            ${{ steps.go-env.outputs.modcache }}
          key: check-generate-${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
          restore-keys: |
            check-generate-${{ runner.os }}-go-
```

Reply:
Now the e2e job takes a minute to download dependencies